From db8d6fd7c8a5ec09d32f15567c5cb1a7aba5df14 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Wed, 5 Jan 2022 15:03:06 +0000
Subject: [PATCH 04/75] HACK: arm64: dts: meson: add LibreELEC device-tree
 overlays for Amlogic

This commit implements a series of device-tree overlays used with LibreELEC
images for Amlogic hardware.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 arch/arm64/boot/dts/amlogic/Makefile             |  2 ++
 arch/arm64/boot/dts/amlogic/overlays/Makefile    |  5 +++++
 .../dts/amlogic/overlays/wifi-rtl8189etv.dts     | 16 ++++++++++++++++
 3 files changed, 23 insertions(+)
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/Makefile
 create mode 100644 arch/arm64/boot/dts/amlogic/overlays/wifi-rtl8189etv.dts

diff --git a/arch/arm64/boot/dts/amlogic/Makefile b/arch/arm64/boot/dts/amlogic/Makefile
index 0eec18678311..34990beb4e5a 100644
--- a/arch/arm64/boot/dts/amlogic/Makefile
+++ b/arch/arm64/boot/dts/amlogic/Makefile
@@ -64,3 +64,5 @@ dtb-$(CONFIG_ARCH_MESON) += meson-sm1-x96-air.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-sm1-x96-air-gbit.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-a1-ad401.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-s4-s805x2-aq222.dtb
+
+subdir-y += overlays
diff --git a/arch/arm64/boot/dts/amlogic/overlays/Makefile b/arch/arm64/boot/dts/amlogic/overlays/Makefile
new file mode 100644
index 000000000000..845b782abcf9
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/Makefile
@@ -0,0 +1,5 @@
+dtbo-y += \
+	wifi-rtl8189etv.dtbo
+
+targets += $(dtbo-y)
+always-y := $(dtbo-y)
diff --git a/arch/arm64/boot/dts/amlogic/overlays/wifi-rtl8189etv.dts b/arch/arm64/boot/dts/amlogic/overlays/wifi-rtl8189etv.dts
new file mode 100644
index 000000000000..a8a0d7d9ab2b
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/overlays/wifi-rtl8189etv.dts
@@ -0,0 +1,16 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@0 {
+		target = <&sd_emmc_a>;
+
+		__overlay__ {
+			/delete-node/ brcmf;
+
+			rtl8189etv: sdio_wifi@1 {
+				reg = <1>;
+			};
+		};
+	};
+};
-- 
2.17.1

